docs: add Collect Prometheus metrics topic for Flow (#3320)
grafana/agent: docs/sources/flow/getting-started/collect-prometheus-metrics.md

4. If your endpoint requires basic authentication, complete the following:

    1. Paste the following inside of the `endpoint` block:

       ```river
       basic_auth {
         username      = "USERNAME"
         password_file = "PASSWORD_FILE"
       }
       ```

    2. Replace `USERNAME` with the basic authentication username to use.

    3. Replace `PASSWORD_FILE` with a path to a file containing the basic
       authentication password or API key.

5. If you have more than one endpoint to write metrics to, repeat the
   `endpoint` block for additional endpoints.

The following example demonstrates configuring `prometheus.remote_write` with
multiple endpoints and mixed usage of basic authentication, and a
`prometheus.scrape` component which forwards metrics to it:

```river
prometheus.remote_write "default" {
  endpoint {
    url = "http://localhost:9090/api/prom/push"
  }

  endpoint {
    url = "https://prometheus-us-central1.grafana.net/api/prom/push"

    basic_auth {
      username      = env("REMOTE_WRITE_USERNAME")
      password_file = "/etc/secrets/api-key"
    }
  }
}

prometheus.scrape "example" {
  targets = [{

>     __address__ = "my-application:80",

Reviewer: Maybe also include a `localhost:12345` target here so that someone trying the example directly can have some metrics flow through?

  }]

  forward_to = [prometheus.remote_write.default.receiver]
}
```

> 

Reviewer: I don't want to make this any more difficult to follow, it's pretty succinct as-is. But do we want a sentence guiding our user to the Flow UI so they can see how data Flows, any debug info, etc?

Author: I don't think so, since we'd probably end up adding it to every single topic (users aren't expected to follow topics in any particular order). But if they follow the first topic, that does point them in the direction of the UI.

Reviewer 2: Just an idea: there could be a "further reading" or "related pages" at the end which list the next things users could be interested in.

Author: We can consider doing that after we've fleshed out the topics more, but I'm also worried that we'll forget to maintain them if we try to keep a list of related topics at the bottom.

---
title: Collect Prometheus metrics
weight: 200
---

# Collect Prometheus metrics

Grafana Agent Flow can be configured to collect [Prometheus][] metrics and
forward them to any Prometheus-compatible database.

This topic describes how to:

* Configure metrics delivery

[Prometheus]: https://prometheus.io

## Components used in this topic

* [prometheus.remote_write][]
* [prometheus.scrape][]

[prometheus.remote_write]: {{< relref "../reference/components/prometheus.remote_write.md" >}}
[prometheus.scrape]: {{< relref "../reference/components/prometheus.scrape.md" >}}

## Before you begin

> * Ensure that you basic familiarity with instrumenting applications with

Reviewer 2: ```suggestion
* Ensure that you have basic familiarity with instrumenting applications with
```

  Prometheus.
* Have a set of Prometheus exports or applications exposing Prometheus metrics
  that you want to collect metrics from.

> * Identify where you will write collected metrics, such as Prometheus, Grafana

Reviewer 2: ```suggestion
* Identify a destination where you will write collected metrics. For example: Prometheus, Grafana
```

Reviewer 2: The current working almost makes it sound like Prometheus is the metric itself.

  Mimir, Grafana Cloud, or Grafana Enterprise Metrics.
* Be familiar with the concept of [Components][] in Grafana Agent Flow.

[Components]: {{< relref "../concepts/components.md" >}}

## Configure metrics delivery

Before components can collect Prometheus metrics, you must have a component
responsible for writing those metrics somewhere.

The [prometheus.remote_write][] component is responsible for delivering
Prometheus metrics to one or Prometheus-compatible endpoints. Once a
`prometheus.remote_write` component is defined, other Grafana Agent Flow
components can be used to forward metrics to it.

To configure a `prometheus.remote_write` component for metrics delivery,
complete the following steps:

1. Add the following `prometheus.remote_write` component to your configuration file:

   ```river
   prometheus.remote_write "LABEL" {
     endpoint {
       url = "PROMETHEUS_URL"
     }
   }
   ```

> 2. Replace `LABEL` with a label to use for the component, such as `default`.

Reviewer 2: It might be more clear if each main point is for adding a component, and there are subpoints for modifying the provided component template. So for example, replacing "LABEL" would be 1.1.
This would make it more consistent with the basic_auth component.

Author: I'm not sure that follows the guidelines for steps, but I'm genuinely not sure. I'll merge as-is and we'll get feedback from the docs team afterwards and clean up if we have to.

   The label chosen must be unique across all `prometheus.remote_write`
   components in the same configuration file.

3. Replace `PROMETHEUS_URL` with the full URL of the Prometheus-compatible
   endpoint where metrics will be sent, such as
   `https://prometheus-us-central1.grafana.net/api/prom/push`.

4. If your endpoint requires basic authentication, complete the following:

>     1. Paste the following inside of the `endpoint` block:

Reviewer 2: Should we call this 4.1, so that we can refer to it more easily if needed? And similar for other sub-points.

Author: No, that won't work in our Markdown.
